org.mwc.debrief.core/src/org/mwc/debrief/core/editors/painters/TimeDisplayPainter.java

```diff
@@ -0,0 +1,642 @@
+ *    modify it under the terms of the Eclipse Public License v1.0
+ *    (http://www.eclipse.org/legal/epl-v10.html)
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ */
+package org.mwc.debrief.core.editors.painters;
+
+import java.awt.Color;
+import java.awt.Point;
+import java.beans.IntrospectionException;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyDescriptor;
+import java.io.Serializable;
+import java.util.Enumeration;
+
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IPartListener;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchPart;
+import org.eclipse.ui.PlatformUI;
+import org.mwc.cmap.core.DataTypes.Temporal.TimeManager;
+import org.mwc.cmap.core.DataTypes.Temporal.TimeProvider;
+import org.mwc.cmap.plotViewer.editors.CorePlotEditor;
+import org.mwc.debrief.core.editors.PlotEditor;
+
+import MWC.GUI.BaseLayer;
+import MWC.GUI.CanvasType;
+import MWC.GUI.Editable;
+import MWC.GUI.ExtendedCanvasType;
+import MWC.GUI.Layers;
+import MWC.GUI.NeedsToBeInformedOfRemove;
+import MWC.GUI.Plottable;
+import MWC.GUI.Properties.DiagonalLocationPropertyEditor;
+import MWC.GenericData.HiResDate;
+import MWC.Utilities.TextFormatting.FormatRNDateTime;
+
+/**
+ * Class to plot a time display onto a plot
+ */
+public class TimeDisplayPainter extends BaseLayer implements NeedsToBeInformedOfRemove
+{
+
+	// ///////////////////////////////////////////////////////////
+	// member variables
+	// //////////////////////////////////////////////////////////
+	/**
+	 * version number for this painter
+	 */
+	static final long serialVersionUID = -1;
+
+	/**
+	 * colour of this time display
+	 */
+	private Color _myColor = new java.awt.Color(28, 228, 28);
+	private Color _myBackgroundColor = new java.awt.Color(128, 128, 128);
+	/**
+	 * whether we are visible or not
+	 */
+	private boolean _isOn = true;
+	
+	private TimeManager _timeManager;
+
+	private CorePlotEditor _thisEditor;
+
+	/**
+	 * default location for the time display
+	 */
+	private int _location = DiagonalLocationPropertyEditor.BOTTOM_RIGHT;
+
+	/**
+	 * our editor
+	 */
+	transient private Editable.EditorType _myEditor;
+
+	private HiResDate _DTG;
+	
+	private String _name = "Time Display";
+	
+	private String _prefix = "";
+	
+	private String _suffix = "";
+	
+	private HiResDate _origin;
+
+	/**
+	 * the font 
+	 */
+	private java.awt.Font _myFont = new java.awt.Font("Arial",
+			java.awt.Font.BOLD, 16);
+	
+	private String _format = "ddHHmm";
+
+	private PropertyChangeListener _timeListener = new PropertyChangeListener()
+	{
+		public void propertyChange(final PropertyChangeEvent event)
+		{
+			// right, retrieve the time
+			_DTG = (HiResDate) event.getNewValue();

```

Comment:
@pecko  - if the user has multiple Time Display objects, won't this cause a screen update for each Time Display object?  So three Time Display objects would cause three screen redraws on each time change...

I'm surprised that our Editable has to listen to time updates, then fire plot updates.  The "highlighters" (normal & snail) already cause a plot redraw when time changes.  Is it possible to "hook into" their processing loop?